src/sentry/search/events/datasets/spans_metrics.py

```diff
@@ -77,6 +77,30 @@ def resolve_metric(self, value: str) -> int:
             self.builder.metric_ids.add(metric_id)
         return metric_id
 
+    def reflective_result_type(
+        self, index: int = 0
+    ) -> Callable[[list[fields.FunctionArg], dict[str, Any]], str]:
+        """
+        Return the type of the metric, defaults to super implementation
+        """
+        super_result_type_fn = super().reflective_result_type(index)
+
+        def result_type_fn(
+            function_arguments: list[fields.FunctionArg], parameter_values: dict[str, Any]
+        ) -> str:
+            argument = function_arguments[index]
+            value = parameter_values[argument.name]
+            mri = constants.SPAN_METRICS_MAP.get(value)
+            if mri is not None:
+                parsed_mri = parse_mri(mri)
+                if parsed_mri is not None and parsed_mri.unit in constants.RESULT_TYPES:
+                    return parsed_mri.unit
+
+            # if the mri is not found, then we fallback to the super implementation
+            return super_result_type_fn(function_arguments, parameter_values)
```

Comment:
Do we need to store the reference to this function, or could we just directly return it here like:
```suggestion
            return super().reflective_result_type(index)(function_arguments, parameter_values)
```

Reply:
Yeah I think that won't work because we're inside a nested function, so `super()` will have the wrong context. But `super(SpansMetricsDatasetConfig, self).reflective...` seems to be the python way of doing it 😄